### PR TITLE
ci: build and test in debug and dev modes

### PR DESCRIPTION
**Pull request by benipeled**
Fix https://github.com/scylladb/scylla-pkg/issues/2767

Verified on https://app.circleci.com/pipelines/github/benipeled/seastar/10/workflows/48ca8e9a-47ad-48cb-8c20-0e8d4eb75033

@xemul @bhalevy

### CODE REVIEW

**Review by bhalevy — CHANGES_REQUESTED** [review]
let's add dev mode too.
alloc failure injection is enabled only in dev mode and not in release nor debug (that doesn't use the seastar allocator)

**Review by bhalevy — CHANGES_REQUESTED** [review]
Te change looks good to me,
just the title should be updated.

**Comment by benipeled** [comment]
dev added
subject updated
Verified with https://app.circleci.com/pipelines/github/benipeled/seastar/11/workflows/fc8ea2a7-96ed-4c8c-b65a-4d20fdd84d4b

**Review by bhalevy — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,8 +22,8 @@ jobs:
       - run:
           command: |
             ./run ./configure.py --compiler << parameters.compiler >> --c++-standard << parameters.standard >>
```
Comment (xemul) [inline .circleci/config.yml:24]: If doing `./configure.py <current stuff> --mode << parameters.mode >>` things get a bit faster
(can be made as a followup)

Comment (benipeled) [reply .circleci/config.yml:24]: make sense, you didn't give me the 15sec to add it :)

**Pull request closed without merging**